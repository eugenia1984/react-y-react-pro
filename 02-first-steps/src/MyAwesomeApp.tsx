import type { CSSProperties } from "react";

interface Address {
  zipCode: string,
  country: string
}

// Al ser constantes que nunca van a cambiar, se pueden dejar fuera del componente
// asi no forman parte del ciclo de renderizado del componente
const firstName: string = 'MarÃ­a Eugenia';
const lastName: string = 'Costa';
const favoriteGames: string[] = ['puzzles', 'mandalas'];
const isActive: boolean = true;
const address: Address = {
  zipCode: 'ABD-123',
  country: 'Spain'
}

const myStyles: CSSProperties = {
  backgroundColor: '#fafafa',
  borderRadius: isActive ? 12 : 8,
  padding: 8
}

export const MyAwesomeApp = () => {
  return (
    <>
      <h1>{lastName}</h1>
      <h2>{firstName}</h2>
      <p>{favoriteGames.join(', ')}</p>
      <p>{isActive ? 'Active' : 'Not  active'}</p>
      <p style={myStyles}> {JSON.stringify(address)} </p>
    </>
  )
}