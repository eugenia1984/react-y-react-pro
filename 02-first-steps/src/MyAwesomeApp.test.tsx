import { describe, expect, test } from "vitest";
import { render, screen } from '@testing-library/react';
import { MyAwesomeApp } from "./MyAwesomeApp";

describe('MyAwesomeApp', () => {
  test('should render firstName and lastName', () => {
    const { container } = render(<MyAwesomeApp />);

    const h1 = container.querySelector('h1'); 
    const h2 = container.querySelector('h2'); 

    expect(h1?.innerHTML).toContain('Costa'); // va a contener
    expect(h2?.innerHTML).toContain('MarÃ­a Eugenia');
  })

  test('should render firstName and lastName - screen', () => {
    render(<MyAwesomeApp />);
    screen.debug();

    const h1 = screen.getByTestId('first-name-title'); 
    const h2 = screen.getByRole('heading', {
      level: 2
    }); 
    
    expect(h1?.innerHTML).toContain('Costa');
    expect(h2?.innerHTML).toContain('Eugenia');
  })
});