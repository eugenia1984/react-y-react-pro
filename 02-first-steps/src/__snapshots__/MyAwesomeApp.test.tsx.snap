// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <h1
    data-testid="first-name-title"
  >
    Costa
  </h1>
  <h2>
    Mar√≠a Eugenia
  </h2>
  <p>
    puzzles, mandalas
  </p>
  <p>
    Active
  </p>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 12px; padding: 8px;"
  >
     
    {"zipCode":"ABD-123","country":"Spain"}
     
  </p>
</div>
`;
